<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>main</class>
 <widget class="QMainWindow" name="main">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1050</width>
    <height>688</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>NEO editor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string/>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QSplitter" name="splitter">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Plain</enum>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="opaqueResize">
          <bool>true</bool>
         </property>
         <property name="childrenCollapsible">
          <bool>true</bool>
         </property>
         <widget class="QTabWidget" name="tabWidget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>250</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::DefaultContextMenu</enum>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab_file">
           <attribute name="title">
            <string>file</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QLineEdit" name="lineEdit_file">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>16777215</height>
               </size>
              </property>
             </widget>
            </item>
            <item>
             <widget class="sourceTree" name="treeWidget_file">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="contextMenuPolicy">
               <enum>Qt::CustomContextMenu</enum>
              </property>
              <property name="itemsExpandable">
               <bool>true</bool>
              </property>
              <property name="headerHidden">
               <bool>true</bool>
              </property>
              <property name="expandsOnDoubleClick">
               <bool>false</bool>
              </property>
              <column>
               <property name="text">
                <string>file</string>
               </property>
              </column>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_elem">
           <attribute name="title">
            <string>data</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QLineEdit" name="lineEdit_data">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item>
             <widget class="sourceTree" name="treeWidget_data">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="mouseTracking">
               <bool>true</bool>
              </property>
              <property name="itemsExpandable">
               <bool>true</bool>
              </property>
              <property name="headerHidden">
               <bool>true</bool>
              </property>
              <property name="expandsOnDoubleClick">
               <bool>false</bool>
              </property>
              <column>
               <property name="text">
                <string>data</string>
               </property>
              </column>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
         <widget class="tabEditor" name="fileEditor">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="focusPolicy">
           <enum>Qt::StrongFocus</enum>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::DefaultContextMenu</enum>
          </property>
          <property name="tabPosition">
           <enum>QTabWidget::North</enum>
          </property>
          <property name="tabShape">
           <enum>QTabWidget::Triangular</enum>
          </property>
          <property name="elideMode">
           <enum>Qt::ElideNone</enum>
          </property>
          <property name="usesScrollButtons">
           <bool>true</bool>
          </property>
          <property name="documentMode">
           <bool>false</bool>
          </property>
          <property name="tabsClosable">
           <bool>true</bool>
          </property>
          <property name="movable">
           <bool>true</bool>
          </property>
          <property name="tabBarAutoHide">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="tabEditor" name="elemEditor">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="focusPolicy">
           <enum>Qt::StrongFocus</enum>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::DefaultContextMenu</enum>
          </property>
          <property name="tabShape">
           <enum>QTabWidget::Triangular</enum>
          </property>
          <property name="currentIndex">
           <number>-1</number>
          </property>
          <property name="elideMode">
           <enum>Qt::ElideNone</enum>
          </property>
          <property name="documentMode">
           <bool>false</bool>
          </property>
          <property name="tabsClosable">
           <bool>true</bool>
          </property>
          <property name="movable">
           <bool>true</bool>
          </property>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1050</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>file</string>
    </property>
    <addaction name="loadProjectAction"/>
    <addaction name="saveProjectAction"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>setting</string>
    </property>
    <addaction name="languageAction"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>function</string>
    </property>
    <addaction name="reloadAction"/>
    <addaction name="recipesAnalysisAction"/>
   </widget>
   <widget class="QMenu" name="menuview">
    <property name="title">
     <string>view</string>
    </property>
    <addaction name="showRecipesAction"/>
    <addaction name="addBrowserAction"/>
    <addaction name="helpAction"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_3"/>
   <addaction name="menuview"/>
   <addaction name="menu_2"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="newProjectAction">
   <property name="text">
    <string>new project</string>
   </property>
   <property name="toolTip">
    <string>new project</string>
   </property>
  </action>
  <action name="loadProjectAction">
   <property name="text">
    <string>load project</string>
   </property>
   <property name="statusTip">
    <string/>
   </property>
  </action>
  <action name="saveProjectAction">
   <property name="text">
    <string>save project</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="reloadAction">
   <property name="text">
    <string>reload</string>
   </property>
   <property name="toolTip">
    <string>reload</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="saveFileAction">
   <property name="text">
    <string>save file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="printFocusAction">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>printFocus</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::WidgetShortcut</enum>
   </property>
  </action>
  <action name="languageAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>change language</string>
   </property>
   <property name="toolTip">
    <string>change to Chinese</string>
   </property>
  </action>
  <action name="helpAction">
   <property name="text">
    <string>help</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="recipesAnalysisAction">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>recipesAnalysis</string>
   </property>
  </action>
  <action name="showRecipesAction">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>showRecipes</string>
   </property>
  </action>
  <action name="addBrowserAction">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>addBrowser</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>tabEditor</class>
   <extends>QTabWidget</extends>
   <header>tabEditor</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>sourceTree</class>
   <extends>QTreeWidget</extends>
   <header>sourceTree</header>
   <slots>
    <slot>filter_file(QString)</slot>
    <slot>show_tooltips(QTreeWidgetItem*)</slot>
    <slot>open_menu(QPoint)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>loadProjectAction</sender>
   <signal>triggered()</signal>
   <receiver>main</receiver>
   <slot>load_project()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>treeWidget_file</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>main</receiver>
   <slot>double_click(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>187</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>695</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>treeWidget_data</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>main</receiver>
   <slot>double_click(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>113</x>
     <y>115</y>
    </hint>
    <hint type="destinationlabel">
     <x>55</x>
     <y>622</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>elemEditor</sender>
   <signal>tabCloseRequested(int)</signal>
   <receiver>main</receiver>
   <slot>remove_data_tab(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>913</x>
     <y>439</y>
    </hint>
    <hint type="destinationlabel">
     <x>1040</x>
     <y>728</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileEditor</sender>
   <signal>tabCloseRequested(int)</signal>
   <receiver>main</receiver>
   <slot>remove_file_tab(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>583</x>
     <y>510</y>
    </hint>
    <hint type="destinationlabel">
     <x>559</x>
     <y>813</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_file</sender>
   <signal>textChanged(QString)</signal>
   <receiver>treeWidget_file</receiver>
   <slot>filter_file(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>113</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>138</x>
     <y>187</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_data</sender>
   <signal>textChanged(QString)</signal>
   <receiver>treeWidget_data</receiver>
   <slot>filter_file(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>113</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>113</x>
     <y>115</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reloadAction</sender>
   <signal>triggered()</signal>
   <receiver>main</receiver>
   <slot>reload()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveProjectAction</sender>
   <signal>triggered()</signal>
   <receiver>main</receiver>
   <slot>save_project()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveFileAction</sender>
   <signal>triggered()</signal>
   <receiver>main</receiver>
   <slot>save_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>languageAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>main</receiver>
   <slot>change_language(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>helpAction</sender>
   <signal>triggered()</signal>
   <receiver>main</receiver>
   <slot>help()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>treeWidget_data</sender>
   <signal>itemEntered(QTreeWidgetItem*,int)</signal>
   <receiver>treeWidget_data</receiver>
   <slot>show_tooltips(QTreeWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>113</x>
     <y>115</y>
    </hint>
    <hint type="destinationlabel">
     <x>113</x>
     <y>115</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>recipesAnalysisAction</sender>
   <signal>triggered()</signal>
   <receiver>main</receiver>
   <slot>recipe_analysis()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showRecipesAction</sender>
   <signal>triggered()</signal>
   <receiver>main</receiver>
   <slot>recipe_show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>treeWidget_file</sender>
   <signal>customContextMenuRequested(QPoint)</signal>
   <receiver>treeWidget_file</receiver>
   <slot>open_menu(QPoint)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>79</x>
     <y>520</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>544</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addBrowserAction</sender>
   <signal>triggered()</signal>
   <receiver>main</receiver>
   <slot>browser_show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>load_project()</slot>
  <slot>save_project()</slot>
  <slot>filter_file(QString)</slot>
  <slot>double_click(QModelIndex)</slot>
  <slot>remove_data_tab(int)</slot>
  <slot>remove_file_tab(int)</slot>
  <slot>show_info(int,int)</slot>
  <slot>reload()</slot>
  <slot>change_language(bool)</slot>
  <slot>save_file()</slot>
  <slot>help()</slot>
  <slot>recipe_analysis()</slot>
  <slot>recipe_show()</slot>
  <slot>browser_show()</slot>
 </slots>
</ui>
